app/services/sendgrid_api.rb

```diff
@@ -1,39 +1,133 @@
 class SendgridApi
-  extend SingleForwardable
+  class Error < StandardError; end
 
-  def_single_delegators :new, :spam_reported?, :bounced?,
-    :remove_from_bounce_list, :remove_from_spam_list
+  TIMEOUT = 2 # seconds
+
+  def initialize
+    @connection = Excon.new(Rails.configuration.sendgrid_api_host,
```

Comment:
I think the configuration parameters (e.g. host) should be passed into the initializer. This code introduces unnecessary coupling with rails & the specific way we store the configuration.